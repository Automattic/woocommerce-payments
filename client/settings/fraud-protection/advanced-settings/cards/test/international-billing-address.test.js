/**
 * External dependencies
 */
import React from 'react';
import { render } from '@testing-library/react';
/**
 * Internal dependencies
 */
import FraudPreventionSettingsContext from '../../context';
import InternationalBillingAddressRuleCard from '../international-billing-address';

describe( 'International billing address card', () => {
	const settings = {
		international_billing_address: {
			enabled: false,
			block: false,
		},
	};
	const setSettings = jest.fn();
	const contextValue = {
		advancedFraudProtectionSettings: settings,
		setAdvancedFraudProtectionSettings: setSettings,
	};
	test( 'renders correctly', () => {
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<InternationalBillingAddressRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
	} );
	test( 'renders correctly when enabled', () => {
		settings.international_billing_address.enabled = true;
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<InternationalBillingAddressRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
	} );
	test( 'renders correctly when enabled and checked', () => {
		settings.international_billing_address.enabled = true;
		settings.international_billing_address.block = true;
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<InternationalBillingAddressRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
	} );
	test( 'renders like disabled when checked, but not enabled', () => {
		settings.international_billing_address.enabled = false;
		settings.international_billing_address.block = true;
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<InternationalBillingAddressRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
	} );
} );
