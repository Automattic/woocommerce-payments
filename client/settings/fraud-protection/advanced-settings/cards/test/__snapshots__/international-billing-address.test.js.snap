// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`International billing address card renders correctly 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection-rule-card css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body fraud-protection-rule-card-header-container css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <p
          class="fraud-protection-rule-card-header"
        >
          International Billing Address
        </p>
        <p
          class="fraud-protection-rule-card-description"
        >
          This filter screens the customer's billing information for addresses outside of your 
          <a
            data-link-type="wc-admin"
            href="#"
          >
            supported countries
          </a>
          .
        </p>
      </div>
    </div>
    <hr />
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        class="fraud-protection-rule-toggle"
      >
        <strong>
          Enable filtering
        </strong>
        <div
          class="components-base-control components-toggle-control fraud-protection-rule-toggle-toggle css-wdf2ti-Wrapper e1puf3u0"
        >
          <div
            class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
          >
            <span
              class="components-form-toggle"
            >
              <input
                aria-describedby="inspector-toggle-control-0__help"
                class="components-form-toggle__input"
                id="inspector-toggle-control-0"
                type="checkbox"
              />
              <span
                class="components-form-toggle__track"
              />
              <span
                class="components-form-toggle__thumb"
              />
            </span>
            <label
              class="components-toggle-control__label"
              for="inspector-toggle-control-0"
            >
              Screen transactions for international billing addresses
            </label>
          </div>
          <p
            class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
            id="inspector-toggle-control-0__help"
          >
            When enabled, the payment method will not be charged until you review and approve the transaction
          </p>
        </div>
      </div>
      <div
        class="fraud-protection-rule-description"
      >
        <strong>
          How does this filter protect me?
        </strong>
        <p>
          Due to the difficulty of authenticating foreign citizens and cross-border legal enforcement against fraudulent activities, you should exercise caution when accepting orders from customers in foreign countries who are using non-domestic addresses.
        </p>
      </div>
      <div
        class="wcpay-banner-notice wcpay-banner-warning-notice fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning components-notice is-warning"
      >
        <div
          class="components-notice__content"
        >
          <div
            class="components-flex css-18koiq9-Flex eboqfv50"
          >
            <div
              class="components-flex__item wcpay-banner-notice__icon wcpay-banner-warning-notice__icon css-1s295sp-Item eboqfv51"
            >
              <svg
                class="gridicon gridicons-notice-outline"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
                  />
                </g>
              </svg>
            </div>
            <div
              class="components-flex__item wcpay-banner-notice__content wcpay-banner-warning-notice__content css-1s295sp-Item eboqfv51"
            >
              Enabling this filter will not have any effect because you are selling to all countries.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`International billing address card renders correctly when enabled 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection-rule-card css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body fraud-protection-rule-card-header-container css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <p
          class="fraud-protection-rule-card-header"
        >
          International Billing Address
        </p>
        <p
          class="fraud-protection-rule-card-description"
        >
          This filter screens the customer's billing information for addresses outside of your 
          <a
            data-link-type="wc-admin"
            href="#"
          >
            supported countries
          </a>
          .
        </p>
      </div>
    </div>
    <hr />
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        class="fraud-protection-rule-toggle"
      >
        <strong>
          Enable filtering
        </strong>
        <div
          class="components-base-control components-toggle-control fraud-protection-rule-toggle-toggle css-wdf2ti-Wrapper e1puf3u0"
        >
          <div
            class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
          >
            <span
              class="components-form-toggle is-checked"
            >
              <input
                aria-describedby="inspector-toggle-control-1__help"
                class="components-form-toggle__input"
                id="inspector-toggle-control-1"
                type="checkbox"
              />
              <span
                class="components-form-toggle__track"
              />
              <span
                class="components-form-toggle__thumb"
              />
            </span>
            <label
              class="components-toggle-control__label"
              for="inspector-toggle-control-1"
            >
              Screen transactions for international billing addresses
            </label>
          </div>
          <p
            class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
            id="inspector-toggle-control-1__help"
          >
            When enabled, the payment method will not be charged until you review and approve the transaction
          </p>
        </div>
        <div>
          <strong>
            Advanced
          </strong>
          <div
            class="fraud-protection-rule-toggle-checkbox-container"
          >
            <div
              class="components-base-control fraud-protection-rule-toggle-checkbox css-wdf2ti-Wrapper e1puf3u0"
            >
              <div
                class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
              >
                <span
                  class="components-checkbox-control__input-container"
                >
                  <input
                    class="components-checkbox-control__input"
                    id="inspector-checkbox-control-0"
                    type="checkbox"
                    value="1"
                  />
                </span>
                <label
                  class="components-checkbox-control__label"
                  for="inspector-checkbox-control-0"
                >
                  Block Payment
                </label>
              </div>
            </div>
            <button
              class="wcpay-tooltip__content-wrapper"
              type="button"
            >
              <div
                class="wcpay-tooltip__content-wrapper"
              >
                <div
                  class="fraud-protection-rule-toggle-checkbox-container-help"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="3 3 18 18"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9.75 10.25C9.75 9.00736 10.7574 8 12 8C13.2426 8 14.25 9.00736 14.25 10.25C14.25 														11.4083 13.3748 12.3621 12.2496 12.4863C12.1124 12.5015 12 12.6119 12 12.75V14M12 														15V16.5M20 12C20 16.4183 16.4183 20 12 20C7.58172 20 4 16.4183 4 12C4 7.58172 														7.58172 4 12 4C16.4183 4 20 7.58172 20 12Z"
                      stroke="#1E1E1E"
                      stroke-width="1.5"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
      <div
        class="fraud-protection-rule-description"
      >
        <strong>
          How does this filter protect me?
        </strong>
        <p>
          Due to the difficulty of authenticating foreign citizens and cross-border legal enforcement against fraudulent activities, you should exercise caution when accepting orders from customers in foreign countries who are using non-domestic addresses.
        </p>
      </div>
      <div
        class="wcpay-banner-notice wcpay-banner-warning-notice fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning components-notice is-warning"
      >
        <div
          class="components-notice__content"
        >
          <div
            class="components-flex css-18koiq9-Flex eboqfv50"
          >
            <div
              class="components-flex__item wcpay-banner-notice__icon wcpay-banner-warning-notice__icon css-1s295sp-Item eboqfv51"
            >
              <svg
                class="gridicon gridicons-notice-outline"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
                  />
                </g>
              </svg>
            </div>
            <div
              class="components-flex__item wcpay-banner-notice__content wcpay-banner-warning-notice__content css-1s295sp-Item eboqfv51"
            >
              Enabling this filter will not have any effect because you are selling to all countries.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`International billing address card renders correctly when enabled and checked 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection-rule-card css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body fraud-protection-rule-card-header-container css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <p
          class="fraud-protection-rule-card-header"
        >
          International Billing Address
        </p>
        <p
          class="fraud-protection-rule-card-description"
        >
          This filter screens the customer's billing information for addresses outside of your 
          <a
            data-link-type="wc-admin"
            href="#"
          >
            supported countries
          </a>
          .
        </p>
      </div>
    </div>
    <hr />
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        class="fraud-protection-rule-toggle"
      >
        <strong>
          Enable filtering
        </strong>
        <div
          class="components-base-control components-toggle-control fraud-protection-rule-toggle-toggle css-wdf2ti-Wrapper e1puf3u0"
        >
          <div
            class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
          >
            <span
              class="components-form-toggle is-checked"
            >
              <input
                aria-describedby="inspector-toggle-control-2__help"
                class="components-form-toggle__input"
                id="inspector-toggle-control-2"
                type="checkbox"
              />
              <span
                class="components-form-toggle__track"
              />
              <span
                class="components-form-toggle__thumb"
              />
            </span>
            <label
              class="components-toggle-control__label"
              for="inspector-toggle-control-2"
            >
              Screen transactions for international billing addresses
            </label>
          </div>
          <p
            class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
            id="inspector-toggle-control-2__help"
          >
            When enabled, the payment method will not be charged until you review and approve the transaction
          </p>
        </div>
        <div>
          <strong>
            Advanced
          </strong>
          <div
            class="fraud-protection-rule-toggle-checkbox-container"
          >
            <div
              class="components-base-control fraud-protection-rule-toggle-checkbox css-wdf2ti-Wrapper e1puf3u0"
            >
              <div
                class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
              >
                <span
                  class="components-checkbox-control__input-container"
                >
                  <input
                    checked=""
                    class="components-checkbox-control__input"
                    id="inspector-checkbox-control-1"
                    type="checkbox"
                    value="1"
                  />
                  <svg
                    aria-hidden="true"
                    class="components-checkbox-control__checked"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M18.3 5.6L9.9 16.9l-4.6-3.4-.9 1.2 5.8 4.3 9.3-12.6z"
                    />
                  </svg>
                </span>
                <label
                  class="components-checkbox-control__label"
                  for="inspector-checkbox-control-1"
                >
                  Block Payment
                </label>
              </div>
            </div>
            <button
              class="wcpay-tooltip__content-wrapper"
              type="button"
            >
              <div
                class="wcpay-tooltip__content-wrapper"
              >
                <div
                  class="fraud-protection-rule-toggle-checkbox-container-help"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="3 3 18 18"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9.75 10.25C9.75 9.00736 10.7574 8 12 8C13.2426 8 14.25 9.00736 14.25 10.25C14.25 														11.4083 13.3748 12.3621 12.2496 12.4863C12.1124 12.5015 12 12.6119 12 12.75V14M12 														15V16.5M20 12C20 16.4183 16.4183 20 12 20C7.58172 20 4 16.4183 4 12C4 7.58172 														7.58172 4 12 4C16.4183 4 20 7.58172 20 12Z"
                      stroke="#1E1E1E"
                      stroke-width="1.5"
                    />
                  </svg>
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
      <div
        class="fraud-protection-rule-description"
      >
        <strong>
          How does this filter protect me?
        </strong>
        <p>
          Due to the difficulty of authenticating foreign citizens and cross-border legal enforcement against fraudulent activities, you should exercise caution when accepting orders from customers in foreign countries who are using non-domestic addresses.
        </p>
      </div>
      <div
        class="wcpay-banner-notice wcpay-banner-warning-notice fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning components-notice is-warning"
      >
        <div
          class="components-notice__content"
        >
          <div
            class="components-flex css-18koiq9-Flex eboqfv50"
          >
            <div
              class="components-flex__item wcpay-banner-notice__icon wcpay-banner-warning-notice__icon css-1s295sp-Item eboqfv51"
            >
              <svg
                class="gridicon gridicons-notice-outline"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
                  />
                </g>
              </svg>
            </div>
            <div
              class="components-flex__item wcpay-banner-notice__content wcpay-banner-warning-notice__content css-1s295sp-Item eboqfv51"
            >
              Enabling this filter will not have any effect because you are selling to all countries.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`International billing address card renders like disabled when checked, but not enabled 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection-rule-card css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body fraud-protection-rule-card-header-container css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <p
          class="fraud-protection-rule-card-header"
        >
          International Billing Address
        </p>
        <p
          class="fraud-protection-rule-card-description"
        >
          This filter screens the customer's billing information for addresses outside of your 
          <a
            data-link-type="wc-admin"
            href="#"
          >
            supported countries
          </a>
          .
        </p>
      </div>
    </div>
    <hr />
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        class="fraud-protection-rule-toggle"
      >
        <strong>
          Enable filtering
        </strong>
        <div
          class="components-base-control components-toggle-control fraud-protection-rule-toggle-toggle css-wdf2ti-Wrapper e1puf3u0"
        >
          <div
            class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
          >
            <span
              class="components-form-toggle"
            >
              <input
                aria-describedby="inspector-toggle-control-3__help"
                class="components-form-toggle__input"
                id="inspector-toggle-control-3"
                type="checkbox"
              />
              <span
                class="components-form-toggle__track"
              />
              <span
                class="components-form-toggle__thumb"
              />
            </span>
            <label
              class="components-toggle-control__label"
              for="inspector-toggle-control-3"
            >
              Screen transactions for international billing addresses
            </label>
          </div>
          <p
            class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
            id="inspector-toggle-control-3__help"
          >
            When enabled, the payment method will not be charged until you review and approve the transaction
          </p>
        </div>
      </div>
      <div
        class="fraud-protection-rule-description"
      >
        <strong>
          How does this filter protect me?
        </strong>
        <p>
          Due to the difficulty of authenticating foreign citizens and cross-border legal enforcement against fraudulent activities, you should exercise caution when accepting orders from customers in foreign countries who are using non-domestic addresses.
        </p>
      </div>
      <div
        class="wcpay-banner-notice wcpay-banner-warning-notice fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning components-notice is-warning"
      >
        <div
          class="components-notice__content"
        >
          <div
            class="components-flex css-18koiq9-Flex eboqfv50"
          >
            <div
              class="components-flex__item wcpay-banner-notice__icon wcpay-banner-warning-notice__icon css-1s295sp-Item eboqfv51"
            >
              <svg
                class="gridicon gridicons-notice-outline"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm1 13h-2v2h2v-2zm-2-2h2l.5-6h-3l.5 6z"
                  />
                </g>
              </svg>
            </div>
            <div
              class="components-flex__item wcpay-banner-notice__content wcpay-banner-warning-notice__content css-1s295sp-Item eboqfv51"
            >
              Enabling this filter will not have any effect because you are selling to all countries.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
